src/highlevel.jl

```diff
@@ -65,9 +65,22 @@ function load_inference(path::AbstractString; execution_provider::Symbol=:cpu,
     if execution_provider === :cpu
         session_options = CreateSessionOptions(api)
     elseif execution_provider === :cuda
-        if !(isdefined(@__MODULE__, :CUDA))
+        CUDAExt = Base.get_extension(@__MODULE__, :CUDAExt)
+        if isnothing(CUDAExt)
             @warn """
```

Comment:
I wonder if we should error instead of warn in these cases.

Reply:
For the first two it's probably a good idea, it will error soon enough anyways. For the last one I'm less certain. https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html#requirements lists CUDA 11.8 as minimum for ONNXRunTime 1.15 but also says it has been tested with 11.6 (a  typo?). Anyway it seems somewhat likely that it can work with a wider range of versions than officially supported.